src/background/listener/bee-api.listener.ts

```diff
@@ -157,122 +178,162 @@ export class BeeApiListener {
+          type: chrome.declarativeNetRequest.RuleActionType.BLOCK,
+        },
+      },
+      {
+        id: BeeApiListener.RESOURCE_LOADER_BLOCKER_ID,
+        priority: 1,
+        condition: {
+          regexFilter: fakeUrl.bzzProtocolRegex,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.BLOCK,
+        },
+      },
+      // Used to load page resources like images
+      // Always have to have session ID in the URL Param
+      {
+        id: BeeApiListener.RESOURCE_LOADER_REDIRECT_ID,
+        priority: 2,
+        condition: {
+          regexFilter: fakeUrl.bzzProtocolRegexWithKey,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.REDIRECT,
+          redirect: {
+            regexSubstitution: `${this._beeApiUrl}/bzz/\\1`,
+          },
+        },
+      },
+      {
+        id: BeeApiListener.BEE_API_BLOCKER_ID,
+        priority: 1,
+        condition: {
+          regexFilter: fakeUrl.beeApiAddressRegex,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.BLOCK,
+        },
+      },
+      {
+        // Redirect the Bee API calls with swarm-session-id query param
+        // The swarm-session-id query parameter can be between the path and the host
+        id: BeeApiListener.BEE_API_REDIRECT_ID,
+        priority: 2,
+        condition: {
+          regexFilter: fakeUrl.beeApiAddressRegexWithKey,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.REDIRECT,
+          redirect: {
+            regexSubstitution: `${this._beeApiUrl}/\\2`,
+          },
+        },
+      },
+      {
+        id: BeeApiListener.BZZ_GOOGLE_BLOCKER_ID,
+        priority: 1,
+        condition: {
+          regexFilter: isHostLocalhost ? bzzSubdomainGoogleRedirectRegex : bzzGoogleRedirectRegex,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.BLOCK,
+        },
+      },
+      // 'bzz://{content-address}' URI in search bar triggers redirect to gateway BZZ address
+      // NOTE: works only if google search is set as default search engine
+      {
+        id: BeeApiListener.BZZ_GOOGLE_REDIRECT_ID,
+        priority: 2,
+        condition: {
+          regexFilter: isHostLocalhost ? bzzSubdomainGoogleRedirectRegex : bzzGoogleRedirectRegex,
+          resourceTypes: BeeApiListener.RESOURCE_TYPE_ALL,
+        },
+        action: {
+          type: chrome.declarativeNetRequest.RuleActionType.REDIRECT,
+          redirect: {
+            regexSubstitution: isHostLocalhost ? `${protocol}://\\1.swarm.${host}\\2` : `${this._beeApiUrl}/bzz/\\1`,
```

Comment:
the subdmain should be in CID form and the bzz path call does not append the original url's path.

Reply:
In these declarative web requests, there is no way for any kind of conversion. How the redirection URL should look like?